app/models/bank_account.rb

```diff
@@ -26,15 +26,17 @@ def bank_and_account_name
 
   # rubocop:disable Naming/PredicateName
   def has_tax_credits?
-    true # TODO: CCMS placeholder - to be fixed by AP-1965
+    tax_credits = bank_transactions.pluck(:meta_data).map { |transaction| transaction[:code] }
+    tax_credits.include?('TC') || tax_credits.include?('WTC') || tax_credits.include?('CTC')
   end
 
   def has_wages?
-    true # TODO: CCMS placeholder - to be fixed by AP-1965
+    false
   end
 
   def has_benefits?
-    true # TODO: CCMS placeholder - to be fixed by AP-1965
+    benefits = bank_transactions.joins(:transaction_type).where(transaction_type: { name: 'benefits' })
+    benefits.present?
```

Comment:
You don't really need the `benefits` variable here. This should work just as well and is a little neater:

```suggestion
    bank_transactions.joins(:transaction_type).where(transaction_type: { name: 'benefits' }).present?
```